pages/api/events.js

```diff
@@ -12,14 +12,29 @@ export default async function handler(req, res) {
   return res.status(200).json(events);
 }
 
-export async function getEvents() {
+export async function getEvents(withLocation = false) {
   let events = [];
   try {
     events = await Profile.aggregate([
       { $project: { username: 1, events: 1, isEnabled: 1 } },
       { $match: { "events.date.start": { $gt: new Date() }, isEnabled: true } },
       { $unwind: "$events" },
       { $match: { "events.date.end": { $gt: new Date() } } },
+      ...(withLocation
+        ? [
+            {
+              $match: {
+                $and: [
+                  { "events.location": { $exists: true } },
+                  { "events.location.lat": { $exists: true } },
+                  { "events.location.lon": { $exists: true } },
+                  { "events.location.lat": { $ne: null } },
+                  { "events.location.lon": { $ne: null } },
+                ],
+              },
+            },
+          ]
+        : []),
```

Comment:
Note to myself to refactor this to something like...

```js
export async function getEvents(withLocation = false) {
  let events = [];
  let aggregate = [
      { $project: { username: 1, events: 1, isEnabled: 1 } },
      { $match: { "events.date.start": { $gt: new Date() }, isEnabled: true } },
      { $unwind: "$events" },
      { $match: { "events.date.end": { $gt: new Date() } } },
      { $sort: { "events.date.start": 1 } },
  ];

  if (withLocation) {
      aggregate.push({
              $match: {
                $and: [
                  { "events.location": { $exists: true } },
                  { "events.location.lat": { $exists: true } },
                  { "events.location.lon": { $exists: true } },
                  { "events.location.lat": { $ne: null } },
                  { "events.location.lon": { $ne: null } },
                ],
              },
            });
  }

  try {
    events = await Profile.aggregate(aggregate)
```

Reply:
Resolved here https://github.com/EddieHubCommunity/LinkFree/pull/8093/commits/04ff78e6197e5d1e83cf1143cf123fcf604b6d0a